internal/web/ui/src/Router.tsx

```diff
@@ -18,7 +18,11 @@ const Router = ({ basePath }: Props) => {
       <main>
         <Routes>
           <Route path="/" element={<PageComponentList />} />
+          <Route path="/remotecfg" element={<PageComponentList />} />
+
           <Route path="/component/*" element={<ComponentDetailPage />} />
+          <Route path="/remotecfg/component/*" element={<ComponentDetailPage />} />
```

Comment:
I tried to reuse the same Page types, by providing a different functionality based on the URL it's called from via `useLocation`.

Let me know if you think that's a good approach or if you'd prefer separate pages here.

Reply:
I don't know anything about web development, but reusing code as a whole is good. My only concern with this approach is that the "Remote Configuration" link in the navigation bar doesn't seem to be highlighted, and it's not clear that the user is actually looking at remote configuration. Is that very hard to fix?

Reply:
I've pushed a commit to reduce code deduplication and also conditionally set the page's name and description (so when you click on Remote Configuration) it looks like this:
![image](https://github.com/user-attachments/assets/33e82259-040e-4168-9efe-ea2ebe5c813d)

Reply:
I think that the cleanest way would be to have different pages but the same components to balance the extensibility and the code duplication. It's possible that in the future we want to add some particular features or designs in the remote config page and wrapping everything with the useRemoteCfg in the common page would add complexity

Reply:
Thanks for the comment, makes sense.

I've added two new pages for the ComponentList and ComponentDetail functionality, which also made them a little simpler (no need to parse the URL itself to figure out whether to useRemoteCfg in https://github.com/grafana/alloy/pull/1472/commits/b3fa5680bea6d25d7c4f033764a2b8c390822028.

The Graph page, I've left as-is for now (the pipelines are not connected in any way _yet_), and the clustering page would be the same for both controllers.